Source/SwiftLintFramework/Rules/Lint/NoBlanketDisablesRule.swift

```diff
@@ -0,0 +1,127 @@
+                violations.append(contentsOf: alreadyDisabledRuleIdentifiers.map {
+                    let reason = "The disabled '\($0.stringRepresentation)' SwiftLint rule was already disabled"
+                    return violation(forPath: file.file.path, command: command, reason: reason)
+                })
+            }
+
+            if command.action == .enable {
+                let notDisabledRuleIdentifiers = command.ruleIdentifiers.subtracting(disabledRuleIdentifiers)
+                violations.append(contentsOf: notDisabledRuleIdentifiers.map {
+                    let reason = "The enabled '\($0.stringRepresentation)' SwiftLint rule was not disabled"
+                    return violation(forPath: file.file.path, command: command, reason: reason)
+                })
+            }
+
+            if command.modifier != nil {
+                continue
+            }
+
+            if command.action == .disable {
+                disabledRuleIdentifiers.formUnion(command.ruleIdentifiers)
+                command.ruleIdentifiers.forEach { ruleIdentifierToCommandMap[$0] = command }
+            }
+            if command.action == .enable {
+                disabledRuleIdentifiers.subtract(command.ruleIdentifiers)
+                command.ruleIdentifiers.forEach { ruleIdentifierToCommandMap.removeValue(forKey: $0) }
+            }
+        }
+
+        let allowedRuleIdentifiers = configuration.allowedRuleIdentifiers
+        for disabledRuleIdentifier in disabledRuleIdentifiers {
+            if allowedRuleIdentifiers.contains(disabledRuleIdentifier.stringRepresentation) {
+                continue
+            }
+
+            if let command = ruleIdentifierToCommandMap[disabledRuleIdentifier] {
+                let reason = "The disabled '\(disabledRuleIdentifier.stringRepresentation)' SwiftLint rule " +
+                             "should be re-enabled before the end of the file"
+                violations.append(violation(forPath: file.file.path, command: command, reason: reason))
+            }
+        }
+
+        violations.append(contentsOf: validateAlwaysBlanketDisable(file: file))
+
+        return violations
+    }
+
+    private func violation(forPath path: String?, command: Command, reason: String) -> StyleViolation {
+        return StyleViolation(
+            ruleDescription: Self.description,
+            severity: configuration.severity,
+            location: Location(file: path, line: command.line, character: command.character),
+            reason: reason
+        )
+    }
+
+    private func validateAlwaysBlanketDisable(file: SwiftLintFile) -> [StyleViolation] {
+        var violations: [StyleViolation] = []
+
+        guard configuration.alwaysBlanketDisableRuleIdentifiers.isEmpty == false else {
+            return []
+        }
+
+        for command in file.commands where command.modifier != nil && command.action == .enable {
```

Comment:
Is `command.action == .enable` correct here?

Are there any tests for this kind of validation?

Reply:
So `command.action == enable` was correct, but `command.modifier != nil` was not :-( . The intent here is that if we specify `always_blanket_disable` for a rule, then the only valid command is `// swiftlint:disable some_rule` - enable, or using a modifier will trigger.

I've improved the reason's to make this clearer (and I wasn't warning on modifiers before, but I think it makes sense - `// swiftlint:disable:next file_length` is almost certainly not intended).

I also added a unit test to cover the `always_blanket_disable` functionality.